linkedin.py

```diff
@@ -161,74 +161,141 @@ def processJob(self, jobID: str, jobCounter: models.JobCounter):
-                
-                textWithinParentheses = utils.extractTextWithinParentheses(full_description)
-                workPlaceType = self.verifyWorkPlaceType(textWithinParentheses)
+                if self.isTitleBlacklisted(jobTitle):
+                    if config.displayWarnings:
+                        prYellow(f"⚠️  Not adding a job as the title '{jobTitle}' is blacklisted")
+                    continue
 
-                if '·' in full_description:
-                    companyName = full_description.split('·')[0].strip()
-                else:
-                    companyName = full_description
+                companyName = self.getCompanyNameFromJobCard(jobItem)
+                if not companyName:
+                    if config.displayWarnings:
+                        prYellow("⚠️  Couldn't extract company name from job card")
+                    continue
 
                 if self.isCompanyBlacklisted(companyName):
                     if config.displayWarnings:
                         prYellow(f"⚠️  Not adding a job as the company '{companyName}' is blacklisted")
                     continue
 
-            if companyName is None:
-                utils.logDebugMessage("Couldn't find companyName", utils.MessageTypes.WARNING)
-                continue
-
-            jobTitleAnchor = jobItem.find_elements(By.XPATH, ".//a[contains(@class, 'job-card-container__link job-card-list__title')]")
-            if len(jobTitleAnchor) > 0:
-                allTexts = jobTitleAnchor[0].text.split("\n")
-                uniqueTexts = list(dict.fromkeys(allTexts))
-                jobTitle = uniqueTexts[0].strip()
-
-                if self.isTitleBlacklisted(jobTitle):
+                jobId = jobItem.get_attribute("data-occludable-job-id")
+                if not jobId:
                     if config.displayWarnings:
-                        prYellow(f"⚠️  Not adding a job as the title '{jobTitle}' is blacklisted")
+                        prYellow("⚠️  Couldn't extract job ID from job card")
                     continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
-                continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
+
+                workPlaceType = self.getWorkplaceTypeFromJobCard(jobItem)
+
+                jobsForVerification.append(models.JobForVerification(
+                    linkedinJobId=jobId.split(":")[-1],
+                    title=jobTitle,
+                    company=companyName,
+                    workplaceType=workPlaceType))
+
+            except Exception as e:
+                if config.displayWarnings:
+                    prYellow(f"⚠️  Error processing job card: {str(e)}")
+                    prYellow(f"Job card HTML: {jobItem.get_attribute('innerHTML')[:200]}...")
```

Comment:
I don't see a valid reason for replacing `if jobTitle is None:` and others with try catch block... Even in those you can use utils.logDebugMessage and log innerHTML. This whole part seems like you went your way instead of following what exists.  

I'm open to discuss your view on this and why this might be a better approach

Reply:
The main reason I initially used try-catch blocks was to handle potential StaleElementReferenceException issues that can occur when working with Selenium. However, you're right that the null checks are cleaner and more explicit. If we do encounter stability issues, we can add error handling at a higher level or implement retries.